### DISCUSSION TITLE
Usage outside of Teams

### DISCUSSION THREAD

**Discussion by kamranayub** [microsoft/teams-sdk#1692] [Q&A]
I know it's in the name but it seems like many of these constructs and behaviors would be awesome to have outside of Teams, like for use in other environments (Slack, or custom web) especially because this is built on top of Bot Framework (which I already use).

Just for context, I have a "legacy" bot I built using LUIS/Cortana Intelligence and Bot Framework (yep), and it used to work on the web, SMS, and even Facebook Messenger. Obviously the next-gen would be creating a Copilot using all these great new services/SDKs, so I'm just trying to understand my options in this brave new world.

If we can't use Teams AI for non-Teams bots, is there another SDK on top of Bot Framework with similar constructs and out-of-the-box behavior? Or is there some undocumented ways to get creative here? 😏

If not, at some point it would be awesome to see a small separation between the abstractions that would work outside Teams for other deployment channels, and then an adapter for Teams-specific features. Like, I'd love to use adaptive cards in my custom React web UI 🔥

I'm wondering if Copilot Studio is closer to handling this ask, but it isn't clear to me if that's the recommendation (and if we miss out on some features).

**Comment by pkbullock** [comment]
Bot supports multiple channels, I for example have built (with preview) using web channel and not used the Teams part. The Teams AI library responds to the bot conversation not to Teams directly.

**Comment by corinagum** [comment]
The majority of use cases from our customers are for Teams scenarios, which is partially why this library was called Teams AI library. For features like Message Extensions, Task Modules, and Teams-specific activities like for meetings, these are only supported in the Teams space (due to being channel specific)

However, the entire AI module is not dependent on Teams in any way, and can be hooked up with other channels. Developers should be able to use the AI features without Teams, it's just not a scenario we provide support for. If you find any AI features that turn out to be channel-specific, that is a bug, and feel free to file one as such.

**Comment by heyitsaamir** [comment]
We're clearing discussions and starting fresh. This discussion has not been acted upon. Please reopen the discussion, or start a new one if this question is still important. Thank you for your understanding.
